src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java

```diff
@@ -79,16 +79,23 @@ public void draw(long vg, int x, int y, InputHandler inputHandler) {
         nanoVGHelper.drawRoundImage(vg, Images.ALPHA_GRID.filePath, x1 + 420, y + 4, 56, 24, 8f, getClass());
         nanoVGHelper.drawRoundedRect(vg, x1 + 420, y + 4, 56, 24, color.getRGB(), 8f);
         if (element.isClicked() && !open) {
-            OneColor finalColor = color;
+            OneColor finalColor = new OneColor(color.getRGB());
             new RenderTickDelay(() -> {
                 open = true;
                 colorSelector = new ColorSelector(finalColor, inputHandler.mouseX(), inputHandler.mouseY(), allowAlpha, inputHandler);
                 OneConfigGui.INSTANCE.initColorSelector(colorSelector);
             }, 1);
         }
         if (OneConfigGui.INSTANCE.currentColorSelector != colorSelector) open = false;
-        else if (open) color = (OneConfigGui.INSTANCE.getColor());
-        setColor(color);
+        else if (open) {
+            try {
+                if (!OneConfigGui.INSTANCE.getColor().toJavaColor().equals(((OneColor) field.get(parent)).toJavaColor()))  {
+                    ((OneColor) field.get(parent)).setFromOneColor(OneConfigGui.INSTANCE.getColor());
```

Comment:
i'm going to assume you have tested this

Reply:
> i'm going to assume you have tested this

I've really tested this i promise

Reply:
okay, can you still change the other issue i pointed out.